flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Expressions.java

```diff
@@ -280,6 +282,40 @@ public static ApiExpression rowInterval(Long rows) {
 		return new ApiExpression(valueLiteral(rows));
 	}
 
+
+	/**
+	 * Creates a SQL INTERVAL literal corresponding to the given {@link Duration}.
+	 * Equivalent to {@code lit(Duration)}.
```

Comment:
this logic is different from `lit(Duration)` because the `ValueDataTypeConverter` analyzes the values and generates suitable intervals: `INTERVAL DAY(2) TO SECOND(3) != INTERVAL SECOND(3)`
Either we don't limit intervals in the API and let the planner fail (this is also what we do in `lit()`) or we call `valueLiteral()` first and check for `INTERVAL SECOND(3)` afterwards.